storm-client/src/jvm/org/apache/storm/utils/NimbusClient.java

```diff
@@ -107,13 +109,14 @@ public static NimbusClient getConfiguredClientAs(Map<String, Object> conf, Strin
             }
             asUser = (String) conf.get(Config.STORM_DO_AS_USER);
         }
+        ConfigValidation.validateFields(conf);
 
-        List<String> seeds;
-        if(conf.containsKey(Config.NIMBUS_HOST)) {
+        List<String> seeds = Lists.newArrayList();
```

Comment:
By default seeds will be empty and will be populated only if NIMBUS_HOST or NIMBUS_SEEDS will be configured. in case if both missing an exception will [be thrown here](https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/utils/NimbusClient.java#L153).